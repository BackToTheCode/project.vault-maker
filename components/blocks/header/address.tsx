/** @jsx jsx */
import { jsx } from '@emotion/core';
import React from 'react';
import { Flex, Box, Text } from 'rebass';

export interface IAddressProps {
  variant?: string;
  children?: React.ReactNode;
  address?: string;
  ern?: string;
  connected?: boolean;
}

const styles = {
  container: {
    width: '100%',
    justifyContent: 'flex-end',
    mr: 20,
    alignItems: 'center'
  },
  indicator: {
    borderRadius: '50%',
    position: 'relative',
    width: '8px',
    height: '8px',
    right: 6,
    top: 6
  },
  addressContainer: {
    px: 4,
    py: 3,
    bg: 'lightGrey',
    borderRadius: '4px'
  }
};

const Address: React.FC<IAddressProps> = ({ connected, address, ern }) => {

  const variant = connected ? 'indicator.connected' : 'indicator.unconnected';

  return (
    <Flex width={1} sx={styles.container}>
      <Box
        sx={{ textAlign: 'left', flexDirection: 'column', maxWidth: '130px' }}
      >
        <Box variant={variant} sx={styles.indicator} />
        <Box sx={styles.addressContainer}>
          <Text
            sx={{ overflow: 'hidden', textOverflow: 'ellipsis' }}
            variant={'body.regular'}
          >
            {address ? address : 'Not connected'}
          </Text>
        </Box>
        {ern && (
          <Box sx={{ marginTop: 2 }}>
            <Text variant={'body.small'} sx={{ color: 'grey' }}>
              {ern}
            </Text>
          </Box>
        )}
      </Box>
    </Flex>
  );
};

export default Address;
