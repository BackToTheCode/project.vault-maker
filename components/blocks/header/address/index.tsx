/** @jsx jsx */
import { jsx } from '@emotion/core';
import React, { FC } from 'react';
import { Box, Flex, Text } from 'rebass';
import styles from './styles';

export interface AddressProps {
  variant?: string;
  children?: React.ReactNode;
  address?: string;
  ern?: string;
  isConnected?: boolean;
}

export const Address: FC<AddressProps> = ({ isConnected, address, ern }) => {
  const variant = isConnected ? 'indicator.connected' : 'indicator.unconnected';

  const connectedMessage = isConnected
    ? address
      ? address
      : 'Connected'
    : 'Not connected';

  return (
    <Flex sx={styles.container}>
      {/* <Box sx={styles.innerContainer}> */}
        <Box variant={variant} sx={styles.indicator} />
        <Box sx={styles.addressContainer}>
          <Text
            sx={styles.address}
            variant={'body.regular'}
          >
            {connectedMessage}
          </Text>
        </Box>
        {ern && (
          <Box sx={styles.ernContainer}>
            <Text variant={'body.small'} sx={styles.ern}>
              {ern}
            </Text>
          </Box>
        )}
      {/* </Box> */}
    </Flex>
  );
};
Address.displayName = 'Address';


