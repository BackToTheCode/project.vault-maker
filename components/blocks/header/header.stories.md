---

The `<Header />` block contains the MakerDAO logo and an address bar that indicates whether you are connected to the Ethereum network and what your public key is.

