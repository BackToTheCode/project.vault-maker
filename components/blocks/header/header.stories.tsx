import { checkA11y } from '@storybook/addon-a11y';
import { boolean, select, withKnobs } from '@storybook/addon-knobs';
import React from 'react';
import { Box } from 'rebass';
import { Header } from '.';
import logoImage from '../../../public/images/mark-maker.svg';

const Background = ({ children }) => (
  <Box sx={{ bg: 'rgb(55,74,97)', ml: 7 }}>{children}</Box>
);

export default {
  component: Header,
  decorators: [checkA11y],
  parameters: {
    backgrounds: [
      { name: 'midGrey', value: 'rgb(55,74,97)', default: true },
      { name: 'primary', value: 'rgb(26, 171, 155)' }
    ]
  },
  title: 'Blocks|Header'
};

export const NotConnected = () => <Header />;

export const Connected = () => (
  <Header
    isConnected={true}
    address="0x61049F5e03Bfe3823f274C479158A94bcA26456c"
  />
);

export const WithKnobs = () => (
  <Header
    isConnected={boolean('isConnected', true)}
    address={select(
      'Address',
      ['0x61049F5e03Bfe3823f274C479158A94bcA26456c', ''],
      '0x61049F5e03Bfe3823f274C479158A94bcA26456c'
    )}
  />
);
WithKnobs.story = {
  decorators: [withKnobs]
};
