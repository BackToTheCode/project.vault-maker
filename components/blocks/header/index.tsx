/** @jsx jsx */
import { jsx } from '@emotion/core';
import React, { Children } from 'react';
import Logo, { ILogoProps } from './logo';
import Address, { IAddressProps } from './address';
import Container from '../../elements/container/index';

export interface IHeaderProps {
  variant?: string;
  children?: React.ReactNode;
  connected?: boolean;
  ern?: string;
  address?: string;
}

type THeader = {
  Logo: React.FC<ILogoProps>;
  Address: React.FC<IAddressProps>;
  Wrapped?: any;
};

const Header: React.FC<IHeaderProps> & THeader = props => {
  const { children, connected } = props;
  return (
    <Container variant="container.wide">
      {Children.map(children, (child: any) => {
        if (child.type.name === 'Logo') {
          return React.cloneElement(child, props);
        }

        if (child.type.name === 'Address') {
          return React.cloneElement(child, { connected });
        }

        return child;
      })}
    </Container>
  );
};

Header.Logo = Logo;
Header.Address = Address;

export default Header
