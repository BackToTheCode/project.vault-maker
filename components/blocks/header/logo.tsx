/** @jsx jsx */
import { jsx } from '@emotion/core';
import { Box } from 'rebass';
import React from 'react';
import { Image } from 'rebass';

export interface ILogoProps {
  variant?: string;
  image: string;
  children?: React.ReactNode;
}

const styles = {
  width: '100%'
};

const Logo: React.FC<ILogoProps> = ({ image }) => (
  <Box width={'100px'}>
    <Image src={image} sx={styles} />
  </Box>
);

export default Logo;
