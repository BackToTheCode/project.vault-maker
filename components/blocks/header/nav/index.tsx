/** @jsx jsx */
import { Button } from '@backtothecode/vm-ui-library';
import { jsx } from '@emotion/core';
import { useRouter } from 'next/router';
import { FC } from 'react';
import { Flex } from 'rebass';
import styles from './styles';

export const Nav: FC<{}> = () => {
    const { container, item, itemActive} = styles;
    const router = useRouter();
    const routeToVaults = (evt: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
        evt.preventDefault();
        router.push('/vaults');
    }

    const routeToMake = (evt: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
        evt.preventDefault();
        router.push('/');
    }

  return (
    <Flex sx={container}>
        <Button onClick={routeToMake} variant={'text'} sx={{...item, ...itemActive}}>Make</Button>
        <Button onClick={routeToVaults} variant={'text'} sx={item}>Vaults</Button>
    </Flex>
  );
};

