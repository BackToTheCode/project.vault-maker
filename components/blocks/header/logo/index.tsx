/** @jsx jsx */
import { jsx } from '@emotion/core';
import { FC } from 'react';
import { Box, Image } from 'rebass';
import logo from '../../../../public/images/mark-maker.svg';
import styles from './styles';

export interface LogoProps {
  variant?: string;
  image?: string;
  children?: React.ReactNode;
}

export const Logo: FC<LogoProps> = ({ image }) => {

  return (
    <Box width={styles.container}>
      <Image src={image} sx={styles.logo} />
    </Box>
  );
};

Logo.defaultProps = {
  image: logo
}
