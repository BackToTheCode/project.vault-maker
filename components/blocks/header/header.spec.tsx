import { render } from '@testing-library/react';
import React from 'react';
import { Header } from '../header';

describe('<Header /> spec', () => {
  test('it renders the component', () => {
    // Arrange
    const expectedText = 'Not connected';

    // Act
    const { getByText } = render(<Header />);

    // Assert
    expect(getByText(expectedText)).toMatchSnapshot();
  });
});
