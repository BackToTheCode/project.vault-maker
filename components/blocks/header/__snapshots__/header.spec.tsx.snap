// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> spec it renders the component 1`] = `
<div
  class="css-6aylez"
>
  Not connected
</div>
`;
